pkg/bff/collaborators.go

```diff
@@ -34,13 +39,33 @@ func (s *Server) AddCollaborator(c *gin.Context) {
 		return
 	}
 
-	// Add the collaborator to the organization
-	if err = org.AddCollaborator(collaborator); err != nil {
-		log.Error().Err(err).Msg("could not add new collaborator to organization")
+	// Make sure the collaborator is valid for storage
+	if err = collaborator.Validate(); err != nil {
+		log.Warn().Err(err).Msg("invalid collaborator in request")
 		c.JSON(http.StatusBadRequest, api.ErrorResponse(err))
 		return
 	}
 
+	// Don't overwrite an existing collaborator
+	id := collaborator.Key()
+	if _, ok := org.Collaborators[id]; ok {
+		log.Warn().Str("collabID", id).Msg("collaborator already exists")
+		c.JSON(http.StatusConflict, api.ErrorResponse("collaborator already exists"))
+		return
+	}
```

Comment:
Now I'm really glad that we're using the md5 hash! It creates an effective uniqueness check that I think will protect us in the long run.